### ISSUE TITLE
Tests fail locally with benchopt version error

### ISSUE THREAD

**Issue by mathurinm** [benchopt/benchopt#182]
If I try to run test_cli.py I get :
```
E AssertionError: Installed the wrong version of benchopt (1.1.1.dev23) in conda env. This should be version: 1.1.1.dev22. There is something wrong the env creation mechanism. Please report this error to https://github.com/benchopt/benchopt
```

This happens here:
```
E           AssertionError: Installed the wrong version of benchopt (1.1.1.dev23) in conda env. This should be version: 1.1.1.dev22. There is something wrong the env creation mechanism. Please report this error to https://github.com/benchopt/benchopt
```

I also get
```
E               RuntimeError: No conda environment is activated.

../cli/main.py:233: RuntimeError
```
in 
```
E               RuntimeError: No conda environment is activated.

../cli/main.py:233: RuntimeError
```

**Comment by tomMoral** [comment]
Is this still an issue?

The first appear to be due to the mismatch between version of editable install. this should have been fixed by #197.
For the second one, I am not sure how this happened?

**Comment by mathurinm** [comment]
I cannot reproduce. `test_cli` pass now, with a few warnings:
```  
/home/mathurin/miniconda3/lib/python3.8/site-packages/click/core.py:2905: DeprecationWarning: 'autocompletion' is renamed to 'shell_complete'. The old name is deprecated and will be removed in Click 8.1. See the docs about 'Parameter' for information about new behavior.
    super().__init__(param_decls, required=required, **attrs)

benchopt/tests/test_cli.py: 1232 warnings
  /home/mathurin/miniconda3/lib/python3.8/site-packages/pip/_vendor/packaging/version.py:111: DeprecationWarning:
  
  Creating a LegacyVersion has been deprecated and will be removed in the next major release

benchopt/tests/test_cli.py::TestInstallCmd::test_benchopt_install_in_env
  /home/mathurin/workspace/benchOpt/benchopt/utils/conda_env_cmd.py:81: UserWarning:
  
  The local version of benchopt (1.1.1.dev58) and the one in conda env (1.1.1.dev67) are different. This can lead to unexpected behavior. You can correct this by either using the --recreate option or fixing the version of benchopt in conda env _benchopt_test_env_c707b105-7837-4132-9ea4-a7bc0d96f8b6.

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=============================== 25 passed, 1252 warnings in 119.96s (0:01:59) =================
```
